lib/quant/series.rb

```diff
@@ -104,20 +104,6 @@ def inspect
       "#<#{self.class.name} symbol=#{symbol} interval=#{interval} ticks=#{ticks.size}>"
     end
 
-    # When the first indicator is instantiated, it will also lead to instantiating
-    # the dominant cycle indicator.  The `new_indicator_lock` prevents reentrant calls
-    # to the `new_indicator` method with infinite recursion.
-    def new_indicator(indicator)
-      return if @new_indicator_lock
-
-      begin
-        @new_indicator_lock = true
-        indicators.new_indicator(indicator)
-      ensure
-        @new_indicator_lock = false
-      end
-    end
-
```

Comment:
This did not work quite as expected.  It prevented the infinite recursion, but fails to add the dominant cycle indicator to the proxy so computations are cached across all indicators.